x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/cohere/CohereEmbeddingsResponseEntity.java

```diff
@@ -188,7 +190,11 @@ private static InferenceServiceResults parseByteEmbeddingsArray(XContentParser p
 
     private static TextEmbeddingByteResults.Embedding parseByteArrayEntry(XContentParser parser) throws IOException {
         XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.currentToken(), parser);
-        List<Byte> embeddingValues = XContentParserUtils.parseList(parser, CohereEmbeddingsResponseEntity::parseEmbeddingInt8Entry);
+        List<Byte> embeddingValuesList = XContentParserUtils.parseList(parser, CohereEmbeddingsResponseEntity::parseEmbeddingInt8Entry);
+        byte[] embeddingValues = new byte[embeddingValuesList.size()];
```

Comment:
Looks like we have this block of code in a few places (more so for floats I think). Should we move the conversion to a static `of` function within `TextEmbeddingByteResults.Embedding` and `TextEmbeddingResults.Embedding`?